chromium_src/third_party/blink/renderer/core/dom/document.cc

```diff
@@ -22,6 +23,31 @@
 
 namespace {
 
+//  Returns the eTLD+1 for the top level frame the document is in.
+//
+//  Returns the eTLD+1 (effective registrable domain) for the top level
+//  frame that the given document is in. This includes frames that
+//  are disconnect, remote or local to the top level frame.
+//
+//  If the top level frame is not discoverable, an empty string is returned.
+std::string topETLDPlusOneForDoc(const Document& doc) {
+  const auto* top_loc_frame = doc.GetFrameOfMasterDocument();
+  if (top_loc_frame == nullptr) {
+    LOG(ERROR) << "Unable to farble frame bc can't find top local frame: "
+        << doc.Url();
+    return "";
+  }
+
+  const auto* top_sec_ctx = top_loc_frame->Tree().Top().GetSecurityContext();
+  if (top_sec_ctx == nullptr) {
+    LOG(ERROR) << "Unable to farble frame bc can't find true top frame: "
+        << doc.Url();
+    return "";
+  }
+
+  return top_sec_ctx->GetSecurityOrigin()->RegistrableDomain().Utf8();
```

Comment:
`RegistrableDomain` is not what we want. See examples in https://url.spec.whatwg.org/#host-registrable-domain

Reply:
fixed!